### PR TITLE
[BugFix] to avoid early deleted shared_ptr in map_apply

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #
1. to avoid early deleted shared_ptr in map_apply, related to https://github.com/StarRocks/starrocks/pull/18979
2. `ASSIGN_OR_RETURN` to replace `EVALUATE_NULL_IF_ERROR`.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/exprs/map_apply_expr.cpp [hunk 1]
```diff
@@ -38,8 +38,9 @@ StatusOr<ColumnPtr> MapApplyExpr::evaluate_checked(ExprContext* context, Chunk*
     std::vector<ColumnPtr> input_columns;
     NullColumnPtr input_null_map = nullptr;
     MapColumn* input_map = nullptr;
+    ColumnPtr input_map_ptr_ref = nullptr; // hold shared_ptr to avoid early deleted.
     // step 1: get input columns from map(key_col, value_col)
-    for (int i = 1; i < _children.size(); ++i) {
+    for (int i = 1; i < _children.size(); ++i) { // currently only 2 children, may be more in the future
         ColumnPtr child_col = EVALUATE_NULL_IF_ERROR(context, _children[i], chunk);
```
Comment (stdpain) [inline be/src/exprs/map_apply_expr.cpp:44]:
```suggestion
        auto child_col = context->evaluate(children[i], chunk);
```

File: be/src/exprs/map_apply_expr.cpp [hunk 2]
```diff
@@ -38,8 +38,9 @@ StatusOr<ColumnPtr> MapApplyExpr::evaluate_checked(ExprContext* context, Chunk*
     std::vector<ColumnPtr> input_columns;
     NullColumnPtr input_null_map = nullptr;
     MapColumn* input_map = nullptr;
+    ColumnPtr input_map_ptr_ref = nullptr; // hold shared_ptr to avoid early deleted.
```
Comment (stdpain) [inline be/src/exprs/map_apply_expr.cpp:41]: captured_input_ptr maybe better ?

File: be/src/exprs/map_apply_expr.cpp (hunk 1 above)
Comment (fzhedu) [reply be/src/exprs/map_apply_expr.cpp]: done, also applied to other related places.

**Comment by fzhedu** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by fzhedu** [comment]
run starrocks_admit_test

**Comment by fzhedu** [comment]
https://github.com/Mergifyio backport branch-3.0

**Comment by fzhedu** [comment]
https://github.com/Mergifyio backport branch-3.0

**Pull request merged**